kanidm_unix_int/src/daemon.rs

```diff
@@ -513,6 +533,36 @@ async fn main() {
                 );
             }
         }
+
+        // check to see if the db's already there
+        if db_path.exists() {
+            if !db_path.is_file() {
+                error!(
+                    "Refusing to run - DB path {} already exists and is not a file.",
+                    db_path.to_str().unwrap_or_else(|| "<db_path invalid>")
+                );
+                std::process::exit(1);
+            };
+
+            let db_metadata = match metadata(&db_path) {
+                Ok(v) => v,
+                Err(e) => {
+                    error!(
+                        "Unable to read metadata for {} - {:?}",
+                        db_path.to_str().unwrap_or_else(|| "<db_path invalid>"),
+                        e
+                    );
+                    std::process::exit(1);
+                }
+            };
+            if db_metadata.permissions().readonly() {
```

Comment:
The rust readonly flag is really bad, so you may not want to check that here. It checks if ALL write bits do not exist, not the write bits for this USER to be able to access it.  So I think it's not really possible to do this.

Reply:
Ack, I re-read the docs and ew. It's possible to do the right dance, but out of scope for this one. :(